src/saml2/client_base.py

```diff
@@ -278,7 +278,13 @@ def is_session_valid(_session_id):
 
     def service_urls(self, binding=BINDING_HTTP_POST):
         _res = self.config.endpoint("assertion_consumer_service", binding, "sp")
-        if _res:
+        if isinstance(_res, tuple):
+            if len(_res) == 1:
+                return _res
+            # if 'assertion_consumer_service' is 3-tuple (URL+binding+index)
```

Comment:
this is the patch that just workrs. I also think that we should investigate better than I actually did, about why we have this behaviour from SPConf